### PR TITLE
Workaround problems when opening solution files in Visual Studio

### PR DESCRIPTION
**Pull request by natemcmaster**
Project reorganization and refactoring has led to a number of problems when opening .sln files. These changes are meant to mitigate issues in using Visual Studio.

Changes:

* Condense Routing.sln into HttpAbstractions.sln
* Workaround NU1105 by adding all ProjectReferences to the .sln
* Workaround exceptions in the ReferencesHostBridge by moving Reference items to a temporary item group
* Add a 'startvs.cmd' script for launching VS with the right env variables
* Remove RangeHelper test project
* Move RangeHelper tests into StaticFiles.Tests and add target for NPM restore

### CODE REVIEW

**Comment by Tratcher** [comment]
I don't think we need the NU1105 sln workaround, I have the build working locally without it. I'll follow up with @JamesNK.

File: src/Middleware/Middleware.sln [hunk 1]
```diff
@@ -135,7 +135,7 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.AspNetCore.Cors",
 EndProject
```
Comment (Tratcher) [inline src/Middleware/Middleware.sln]: We should remove the middleware.sln. It's already large enough that VS is having trouble loading it. It's also not useful as none of these projects reference eachother or are worked on together. Each middleware is it's own feature area and could have its own sln.

Comment (natemcmaster) [reply src/Middleware/Middleware.sln]: Can we treat this as a separate issue?

Comment (Tratcher) [reply src/Middleware/Middleware.sln]: Sure

**Comment by Tratcher** [comment]
@JamesNK wasn't able to get it working following the same steps I used, not sure why. We can try the sln workaround.

File: docs/BuildFromSource.md [hunk 2]
```diff
@@ -58,22 +53,66 @@ git submodule update --init --recursive
...
-Before opening our .sln files in Visual Studio or VS Code, executing the following on command-line:
-```
-.\build.cmd /t:Restore
-```
-This will download required tools.
+Before opening our .sln files in Visual Studio or VS Code, you need to perform the following actions.
+
+1. Executing the following on command-line:
+   ```
+   .\build.cmd /p:SkipTests=true /p:_ProjectsOnly=true
```
Comment (Tratcher) [inline docs/BuildFromSource.md]: `/p:BuildNumberSuffix=t9999`, or have you addressed that another way?

**Review by Tratcher — APPROVED** [review]

File: docs/BuildFromSource.md (hunk 2 above)
Comment (natemcmaster) [reply docs/BuildFromSource.md]: `/p:BuildNumberSuffix=t9999` is only necessary in the master branch, but should be resolved by #4311. I'll add that to docs when I merge this up thru 2.2 and master.

**Comment by JamesNK** [comment]
I tried out building in the `release/2.1` branch and there is one NU1105 error remaining for `<ProjectReference Include="..\testassets\RoutingSample.Web\RoutingSample.Web.csproj" />`. The sample website isn't part of the solution.

**Comment by natemcmaster** [comment]
Which solution?

**Comment by JamesNK** [comment]
Ah, right: `\src\Http\HttpAbstractions.sln`

**Comment by natemcmaster** [comment]
Yeah, I see that now too. I'll add that to the .sln in my next PR. It seems NU1105 may not appear if you have already restored the projects.

**Pull request merged**